conans/client/remote_manager.py

```diff
@@ -127,21 +127,22 @@ def compress_files(files, name, excluded):
     tgz_contents = StringIO()
     tgz = gzopen_without_timestamps(name, mode="w", fileobj=tgz_contents)
 
-    def addfile(name, contents, tar):
+    def addfile(name, file_info, tar):
         info = tarfile.TarInfo(name=name)
-        string = StringIO(contents)
-        info.size = len(contents)
+        string = StringIO(file_info["contents"])
+        info.size = len(file_info["contents"])
```

Comment:
This is where the change actually occurs - a dict is now passed to compress_files which contains both the contents and the mode of the file. Maybe it could include more metadata in future depending on what is needed to be retained.